lgc/builder/InOutBuilder.cpp

```diff
@@ -417,23 +423,26 @@ void BuilderImpl::markGenericInputOutputUsage(bool isOutput, unsigned location,
         keepAllLocations = true;
     }
     unsigned startLocation = (keepAllLocations ? 0 : location);
+    // NOTE: The non-invalid value as initial new Location info or new location is used to identify the dynamic indexing
+    // location.
+    unsigned initialVal = (isDynLocOffset && locationCount > 1) ? location : InvalidValue;
     // Non-GS-output case.
     if (inOutLocInfoMap) {
       for (unsigned i = startLocation; i < location + locationCount; ++i) {
         InOutLocationInfo origLocationInfo;
         origLocationInfo.setLocation(i);
         origLocationInfo.setComponent(inOutInfo.getComponent());
         auto &newLocationInfo = (*inOutLocInfoMap)[origLocationInfo];
-        newLocationInfo.setData(InvalidValue);
+        newLocationInfo.setData(initialVal);
```

Comment:
Why isn't this offset somehow? That is to say, if locationCount > 1, why do all of the original locations get mapped to the same location here? I would have rather expected something like:
```
  newLocationInfo.setData(forceDynamicUse ? i : InvalidValue);
```

Reply:
There is no special reason to map to the same location for dynamic location offset. I just want to make it different from InvalidValue in order to identify this location is caculated based on an offset later. Changed to your perfered shape.